### ISSUE TITLE
Grade school exercise will not properly sort grades beyond :grade_9

### ISSUE THREAD

**Issue by parkerl** [exercism/xelixir#109]
I'm afraid we've swapped on bug for another. Using atoms for the grade levels has the unfortunate side effect that sorting is done like a string rather than a number. That means `:grade_11` will come before `:grade_2`. See https://github.com/parkerl/xelixir/commit/ce61f421c0399d458d87c4d4a9459253b7c311b4 

This fails:

```
1) test sort school by grade and by student name (SchoolTest)
     grade_school_test.exs:55
     Assertion with == failed
     code: expected == actual
     lhs:  [grade_3: ["Kyle"], grade_4: ["Bart", "Christopher", "Jennifer"], grade_6: ["Kareem"], grade_11: ["John"]]
     rhs:  [grade_11: ["John"], grade_3: ["Kyle"], grade_4: ["Bart", "Christopher", "Jennifer"], grade_6: ["Kareem"]]
     stacktrace:
       grade_school_test.exs:72
```
My thought would be to revert to the original implementation but have `sort` return a list of two tuples with the grade and names, `[{1, ["John"]}]`. This would leave most of the original exercise intact and only change the api of the function with the bug. 

I'm sorry I was too slow to get this review prepared before the PR was merged. My bad. 

\\cc @bordeltabernacle

**Comment by bordeltabernacle** [comment]
Could we not just change `:grade_2` to `:grade_02`?

```
iex(3)> Enum.sort([:grade_11, :grade_02])
[:grade_02, :grade_11]
iex(4)> Enum.sort([:grade_2, :grade_11]) 
[:grade_11, :grade_2]
```

Sorry I missed this.

**Comment by rubysolo** [comment]
Zero-padding inside a keyword feels like a hack, IMO.

I do like the explicitness of the `:grade` prefix -- what about using a tagged tuple for the grade?

```elixir
iex(1)> Enum.sort([{:grade, 11}, {:grade, 2}])
[grade: 2, grade: 11]
iex(2)> Enum.sort([{{:grade, 11}, []}, {{:grade, 2}, []}])
[{{:grade, 2}, []}, {{:grade, 11}, []}]
```

It's a bit noisy, but (I think) a bit more Erlang-y.

**Comment by bordeltabernacle** [comment]
Am I right in thinking this would mean the `db` is just an ordinary list, rather than a `keyword list`, and therefore the data in it is only accessible by index or by recursion/map/filter through the list until you get a pattern match on, say, grade? ie.

```elixir
iex(29)> db |> Enum.filter(fn({{:grade, x}, y}) -> x == 2 end)
[{{:grade, 2}, ["Aimee"]}]  
```

How would you add multiple people to a grade? It seems to me that that would be a different operation to adding the first person to an empty grade, or to an empty `db`, unless you had the `db` structure already in place?

Could a list of maps be used perhaps?

```elixir
db = [%{1 => ["Aimee"]}, %{2 => ["John", "Matt"]}]
```

I'm probably missing something, or overthinking this, but I'm having a hard time understanding how you would operate on this if it's not a `keyword list` or a `map`.  I don't have any Erlang experience, but afaik using a `keyword list` or a `map` would be the Elixir-y way to approach the problem?

**Comment by rubysolo** [comment]
I think keyword list or map work well for the internal representation of the data, but maybe not as well for the value returned by `sort`, since it obviously needs to maintain sort order.

**Comment by bordeltabernacle** [comment]
Thinking about this again today, I agree with @parkerl that reverting back to using a map but have the sort return a two-tuple list, is probably the best resolution to this.  I see what you mean now @rubysolo about separating (& differentiating between) the internal representation of the data and what is returned by a function, such as `sort`. 
I'm happy to make the changes (undo my mistake :grin: ) depending on what you both think.

**Comment by rubysolo** [comment]
:+1:

**Comment by parkerl** [comment]
Closed by #118
